packages/react-components/react-rating-preview/src/contexts/RatingContext.tsx

```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+
+import type { RatingContextValue } from '../Rating';
+
+/**
+ * RadioGroupContext is provided by RadioGroup, and is consumed by Radio to determine default values of some props.
+ */
+export const RatingContext = React.createContext<RatingContextValue | undefined>(undefined);
+
+const defaultIcon: React.ReactElement = <div />;
+const ratingContextDefaultValue: RatingContextValue = {
+  appearance: 'outline',
+  compact: false,
+  iconFilled: defaultIcon,
+  iconOutline: defaultIcon,
+  name: '',
+  precision: false,
+  readOnly: false,
+  size: 'medium',
+  value: 0,
+};
+
+export const RatingProvider = RatingContext.Provider;
+
+/**
+ * Get the value of the RadioGroupContext.
+ */
+export const useRatingContextValue_unstable = () => React.useContext(RatingContext) || ratingContextDefaultValue;
```

Comment:
The default value for RatingContext can just be `undefined` to simplify this, since you're not relying on the default value anywhere. No need to go to the effort to generate default values for these.
```suggestion
export const RatingProvider = RatingContext.Provider;

/**
 * Get the value of the RadioGroupContext.
 */
export const useRatingContextValue_unstable = () => React.useContext(RatingContext);
```